### PR TITLE
dpg, deviceupdate, support relative url

### PR DESCRIPTION
**Pull request by haolingdong-msft**
# Description

The code is updated based on core pr: https://github.com/Azure/azure-sdk-for-java/pull/29676. Since core pr is not merged and released, we need to copy code to the `implementation` folder in this pr.

# All SDK Contribution checklist:
- [ ] **The pull request does not introduce [breaking changes]**
- [ ] **CHANGELOG is updated for new features, bug fixes or other significant changes.**
- [ ] **I have read the [contribution guidelines](https://github.com/Azure/azure-sdk-for-java/blob/main/CONTRIBUTING.md).**

## [General Guidelines and Best Practices](https://github.com/Azure/azure-sdk-for-java/blob/main/CONTRIBUTING.md#developer-guide)
- [ ] Title of the pull request is clear and informative.
- [ ] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).

### [Testing Guidelines](https://github.com/Azure/azure-sdk-for-java/blob/main/CONTRIBUTING.md#building-and-unit-testing)
- [ ] Pull request includes test coverage for the included changes.

### CODE REVIEW

File: sdk/deviceupdate/azure-iot-deviceupdate/src/test/java/com/azure/iot/deviceupdate/DeviceUpdateClientTests.java [hunk 1]
```diff
@@ -133,4 +137,13 @@ public void testGetFilesNotFound() {
             assertEquals(404, e.getResponse().getStatusCode());
         }
     }
+
+    @Disabled("session records not ready")
+    @Test
+    public void testImportUpdates() {
```
Comment (haolingdong-msft) [inline sdk/deviceupdate/azure-iot-deviceupdate/src/test/java/com/azure/iot/deviceupdate/DeviceUpdateClientTests.java:143]: This test can be used to record session-records.

File: sdk/deviceupdate/azure-iot-deviceupdate/src/test/java/com/azure/iot/deviceupdate/TestData.java [hunk 2]
```diff
@@ -10,7 +10,7 @@ public class TestData {
     public static final String CLIENT_ID = getTestValue("CLIENT_ID", "clientId");
 
     public static final String ACCOUNT_ENDPOINT = getTestValue("ACCOUNT_ENDPOINT",
-        "contosoprod.api.prod.adu.microsoft.com");
+        "https://adu-sdk-testing.api.adu.microsoft.com");
```
Comment (weidongxu-microsoft) [inline sdk/deviceupdate/azure-iot-deviceupdate/src/test/java/com/azure/iot/deviceupdate/TestData.java:13]: why this change?

File: sdk/deviceupdate/azure-iot-deviceupdate/src/main/java/com/azure/iot/deviceupdate/implementation/PollingUtils.java [hunk 3]
```diff
@@ -0,0 +1,108 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
```
Comment (weidongxu-microsoft) [inline sdk/deviceupdate/azure-iot-deviceupdate/src/main/java/com/azure/iot/deviceupdate/implementation/PollingUtils.java:1]: actually most of this class is likely not needed, since you only need to handle `BinaryData` (i.e., U and T in poller is just `BinaryData`).

but fine to keep the logic simple and keep the poller class unchanged.

**Review by weidongxu-microsoft — COMMENTED** [review]
LGTM

when ready, add David to reviewer.

File: sdk/deviceupdate/azure-iot-deviceupdate/src/main/java/com/azure/iot/deviceupdate/implementation/PollingUtils.java (hunk 3 above)
Comment (weidongxu-microsoft) [reply sdk/deviceupdate/azure-iot-deviceupdate/src/main/java/com/azure/iot/deviceupdate/implementation/PollingUtils.java:1]: but if you need to suppress in spotbugs, personally I prefer just delete them.

File: sdk/deviceupdate/azure-iot-deviceupdate/src/test/java/com/azure/iot/deviceupdate/TestData.java (hunk 2 above)
Comment (haolingdong-msft) [reply sdk/deviceupdate/azure-iot-deviceupdate/src/test/java/com/azure/iot/deviceupdate/TestData.java:13]: because endpoint needs to be a url (start with http) so that it can concat with relative url.

Comment (weidongxu-microsoft) [reply sdk/deviceupdate/azure-iot-deviceupdate/src/test/java/com/azure/iot/deviceupdate/TestData.java:13]: then you are wrong here.

see the host is composed based on the endpoint https://github.com/Azure/azure-sdk-for-java/blob/main/sdk/deviceupdate/azure-iot-deviceupdate/src/main/java/com/azure/iot/deviceupdate/implementation/DeviceManagementsImpl.java#L71

We might need to think how we should handle this in codegen on full-auto. But here apparently you should have `"https://" + this.client.getEndpoint()`

File: sdk/deviceupdate/azure-iot-deviceupdate/src/main/java/com/azure/iot/deviceupdate/implementation/PollingUtils.java (hunk 3 above)
Comment (dpokluda) [reply sdk/deviceupdate/azure-iot-deviceupdate/src/main/java/com/azure/iot/deviceupdate/implementation/PollingUtils.java:1]: All this code is autogenerated. I am not going to touch it. (if it was question for me).

**Comment by dpokluda** [comment]
I don't fully follow what you are doing here. I need my LRO to work with relative urls. If this change will allow that, then I am approving this change.

**Review by dpokluda — APPROVED** [review]

**Comment by haolingdong-msft** [comment]
> I don't fully follow what you are doing here. I need my LRO to work with relative urls. If this change will allow that, then I am approving this change.

Hi @dpokluda , this pr supports LRO with relative urls, you use this test to do a LIVE test and records session-records if you want. https://github.com/Azure/azure-sdk-for-java/blob/2512c1f5f14def3705371b19b773c399194286cc/sdk/deviceupdate/azure-iot-deviceupdate/src/test/java/com/azure/iot/deviceupdate/DeviceUpdateClientTests.java#L145

**Comment by dpokluda** [comment]
Excellent, thank you very much. :-)

**Review by weidongxu-microsoft — APPROVED** [review]

**Review by XiaofeiCao — APPROVED** [review]
